src/Microsoft.ML.Data/Transforms/KeyToVectorTransform.cs

```diff
@@ -708,10 +708,26 @@ private JToken SaveAsPfaCore(BoundPfaContext ctx, int iinfo, ColInfo info, JToke
 
             private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, ColInfo info, string srcVariableName, string dstVariableName)
             {
+                var shape = ctx.RetrieveShape(srcVariableName);
+
+                // If Bag is true, the output of ONNX LabelEncoder needs to be fed into ONNX ReduceSum because
+                // default ONNX LabelEncoder just matches the behavior of Bag=false.
+                var encodedVariableName = _parent._columns[iinfo].Bag ? ctx.AddVariable("encoded") : dstVariableName;
+
                 string opType = "OneHotEncoder";
-                var node = ctx.CreateNode(opType, srcVariableName, dstVariableName, ctx.GetNodeName(opType));
+                var node = ctx.CreateNode(opType, srcVariableName, encodedVariableName, ctx.GetNodeName(opType));
```

Comment:
Can we have a test which cover this case?
Otherwise, chance of degradation is quite high.

Reply:
May we do it later? The current conversion framework doesn't work with new API.

Reply:
A test which checks if bag+ leads to the correct ONNX pattern has been added.